embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryServiceTest.kt

```diff
@@ -2,67 +2,61 @@ package io.embrace.android.embracesdk.comms.delivery
+import io.embrace.android.embracesdk.fakes.FakeApiService
+import io.embrace.android.embracesdk.fakes.FakeDeliveryCacheManager
 import io.embrace.android.embracesdk.fakes.FakeGatingService
 import io.embrace.android.embracesdk.fakes.fakeSession
 import io.embrace.android.embracesdk.fakes.fakeSessionMessage
 import io.embrace.android.embracesdk.internal.serialization.EmbraceSerializer
 import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
-import io.embrace.android.embracesdk.ndk.NdkService
 import io.embrace.android.embracesdk.payload.Event
 import io.embrace.android.embracesdk.payload.EventMessage
 import io.embrace.android.embracesdk.session.SessionSnapshotType.JVM_CRASH
 import io.embrace.android.embracesdk.session.SessionSnapshotType.NORMAL_END
 import io.embrace.android.embracesdk.session.SessionSnapshotType.PERIODIC_CACHE
-import io.mockk.Called
-import io.mockk.clearAllMocks
-import io.mockk.every
-import io.mockk.mockk
-import io.mockk.verify
 import org.junit.After
 import org.junit.Assert.assertEquals
-import org.junit.Assert.assertNotNull
-import org.junit.BeforeClass
+import org.junit.Assert.assertTrue
+import org.junit.Before
 import org.junit.Test
-import java.util.concurrent.Future
-import java.util.concurrent.TimeUnit
+import java.util.concurrent.ExecutorService
 
 internal class EmbraceDeliveryServiceTest {
 
-    private lateinit var deliveryService: EmbraceDeliveryService
-    private val executor = MoreExecutors.newDirectExecutorService()
     private val session = fakeSession()
     private val sessionMessage = fakeSessionMessage()
+    private val anotherMessage =
+        fakeSessionMessage().copy(session = session.copy(sessionId = "session2"))
+
+    private lateinit var executor: ExecutorService
+    private lateinit var deliveryCacheManager: FakeDeliveryCacheManager
+    private lateinit var apiService: FakeApiService
+    private lateinit var ndkService: FakeNdkService
+    private lateinit var gatingService: FakeGatingService
+    private lateinit var logger: InternalEmbraceLogger
+    private lateinit var deliveryService: EmbraceDeliveryService
 
-    companion object {
-        private lateinit var mockDeliveryCacheManager: EmbraceDeliveryCacheManager
-        private lateinit var apiService: ApiService
-        private lateinit var gatingService: FakeGatingService
-        private lateinit var logger: InternalEmbraceLogger
-
-        @BeforeClass
-        @JvmStatic
-        fun beforeClass() {
-            mockDeliveryCacheManager = mockk(relaxed = true)
-            every { mockDeliveryCacheManager.loadCrash() } returns null
-            every { mockDeliveryCacheManager.getAllCachedSessionIds() } returns emptyList()
-            apiService = mockk(relaxed = true)
-            gatingService = FakeGatingService()
-            logger = InternalEmbraceLogger()
-        }
+    @Before
+    fun setUp() {
+        executor = MoreExecutors.newDirectExecutorService()
```

Comment:
Isn't it better to use BlockingScheduledExecutorService?

Reply:
I'll leave this as-is since I was just moving the point at which it's initialized, but in future perhaps we can do a roundup of anything using `MoreExecutors`